### PR TITLE
Refactor public API

### PR DESCRIPTION
**Pull request by LordTermor**
This PR refactors and cleans up the whole public API by streamlining response creation and removing hand-crafted serialization in favour of `reflect-cpp` library and struct-based requests and responses.

@romangg code-wise it's complete and tested, only the OpenAPI config is not updated to represent these changes so feel free to review it

@fhdk this will break public API so keep an eye on it for the CLI version

### CODE REVIEW

File: daemon/presentation/responses/UserResponse.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+#include <set>
+#include <string>
+
+namespace bxt::Presentation {
+struct UserResponse {
+    std::string name;
+    std::set<std::string> permissions;
+};
+} // namespace bxt::Presentation
```
[the remark this answers is not in the document]
Comment (LordTermor) [reply daemon/presentation/responses/UserResponse.h:17]: Whatever you prefer

[the remark this answers is not in the document]
Comment (romangg) [reply daemon/presentation/responses/UserResponse.h:17]: If there is no reason for splitting them up per individual request I prefer to put together all responnses in one header and the requests in one. Thanks.

File: daemon/core/domain/enums/LogEntryType.h [hunk 2]
```diff
@@ -8,6 +8,6 @@
 
 namespace bxt::Core::Domain {
 
-enum LogEntryType { Add, Remove, Update };
+enum struct LogEntryType { Add, Remove, Update };
```
[the remark this answers is not in the document]
Comment (LordTermor) [reply daemon/core/domain/enums/LogEntryType.h]: > enum class is more common
Oh. Yeah.

> Why is this part of the commit?

For whatever reason reflect-cpp doesn't allow to serialize LogEntryType if it's not marked as struct/class

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.23)
```
[the remark this answers is not in the document]
Comment (LordTermor) [reply CMakeLists.txt:1]: Done.

File: daemon/utilities/drogon/Helpers.h [hunk 4]
```diff
@@ -39,4 +44,20 @@ inline ::drogon::HttpResponsePtr
...
+    result->setStatusCode(drogon::k200OK);
+
+    return result;
+}
+
+template<typename T> inline auto get_request_json(drogon::HttpRequestPtr req) {
+    return rfl::json::read<T, rfl::SnakeCaseToCamelCase>(
+        std::string(req->body()));
+}
+
```
[the remark this answers is not in the document]
Comment (LordTermor) [reply daemon/utilities/drogon/Helpers.h:62]: Fixed.

File: daemon/presentation/responses/UserResponse.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (LordTermor) [reply daemon/presentation/responses/UserResponse.h]: I've split them so now they match the controllers, it looks better and it's easier to find this way imo

**Review by romangg — APPROVED** [review]

**Pull request merged**